addons/account/models/account_move.py

```diff
@@ -2900,12 +2900,32 @@ def _prepare_invoice_aggregated_taxes(self, filter_invl_to_apply=None, filter_ta
             to_update_vals, tax_values_list = self.env['account.tax']._compute_taxes_for_single_line(base_line)
             to_process.append((base_line, to_update_vals, tax_values_list))
 
-        return self.env['account.tax']._aggregate_taxes(
+        tax_details = self.env['account.tax']._aggregate_taxes(
             to_process,
             filter_tax_values_to_apply=filter_tax_values_to_apply,
             grouping_key_generator=grouping_key_generator,
         )
 
+        # Handle manually changed tax amounts (via quick-edit or journal entry manipulation)
+        tax_lines = self.line_ids.filtered(lambda x: x.display_type == 'tax')
+        if tax_lines is not None:
+            def _in_company_currency(value):
+                return self.currency_id._convert(value, self.company_currency_id, self.company_id, self.invoice_date)
+            tax_line_values = [x._convert_to_tax_line_dict() for x in tax_lines]
+            for tax_detail in tax_details['tax_details'].values():
+                tax_ids = [tax['id'] for tax in tax_detail['group_tax_details']]
+                matched_tax_lines = [
+                    x for x in tax_line_values
+                    if x['tax_repartition_line'].tax_id.id in tax_ids
+                ]
+                if matched_tax_lines:
+                    tax_detail['tax_amount_currency'] = sum(x['tax_amount'] for x in matched_tax_lines)
+                    tax_detail['tax_amount'] = _in_company_currency(tax_details['tax_amount_currency'])
+            tax_details['tax_amount_currency'] = sum(tax_detail['tax_amount_currency'] for tax_detail in tax_details['tax_details'].values())
+            tax_details['tax_amount'] = _in_company_currency(tax_details['tax_amount_currency'])
```

Comment:
It's too late to modify the dictionary after `_aggregate_taxes`. Instead, distribute the difference accross the `tax_values_list` in `to_process`. That way, you ensure a consistency accross all values in any level.
Also, take care of distributing the difference smoothly. If you have a difference of 0.03, don't put all of them on the same tax details but dispatch 0.01 on 3 differents tax details